src/core/control/jobs/AutosaveJob.cpp

```diff
@@ -30,10 +30,7 @@ void AutosaveJob::run() {
 
     Document* doc = control->getDocument();
 
-    doc->lock();
-    handler.prepareSave(doc);
     auto filepath = doc->getFilepath();
```

Comment:
I think `doc->getFilepath();` still needs the mutex lock (imagine the user renames the file -- or saves as -- while the autosave fetchs the filepath.